.travis.yml

```diff
@@ -2,24 +2,27 @@ language: generic
 services:
   - docker
 
-notifications:
-  email:
-    on_success: always
-    on_failure: always
+cache:
+  directories:
+    - $HOME/.ccache
+
+git:
+  quiet: true
+
 env:
+  global:
+    - CCACHE_DIR=$HOME/.ccache
+    - ROS_REPO=ros
   matrix:
-    - ROS_DISTRO="kinetic"   ROS_REPO=ros
-    - ROS_DISTRO="kinetic"   ROS_REPO=ros-shadow-fixed
-    - ROS_DISTRO="kinetic"   PRERELEASE=true
-    - ROS_DISTRO="melodic"   ROS_REPO=ros
-    - ROS_DISTRO="melodic"   ROS_REPO=ros-shadow-fixed
-    - ROS_DISTRO="melodic"   PRERELEASE=true
-matrix:
-  allow_failures:
-    - env: ROS_DISTRO="melodic"   ROS_REPO=ros
-    - env: ROS_DISTRO="melodic"   ROS_REPO=ros-shadow-fixed
-    - env: ROS_DISTRO="melodic"   PRERELEASE=true
+    - ROS_DISTRO="kinetic"
+    - ROS_DISTRO="kinetic" ROS_REPO=ros-testing
```

Comment:
`ROS_REPO=[ros-]testing` is the default, you would run the same job twice.
To use the stable release, you have to set `ROS_REPO=ros` (even better: `ROS_REPO=main`) explicitly.
If you want to, you can mark the others with `ROS_REPO=testing` for readability.

Reply:
Isn' that done by the `ROS_REPO=ros` in the `global` section?

Reply:
Ah yes, you are right. I did not see that line.
I would remove it as a global. You are adding both variants anyway.